templates/common/infra/bicep/core/host/aks/firewall.bicep

```diff
@@ -0,0 +1,324 @@
+param resourceName string
+param location string = resourceGroup().location
+param workspaceDiagsId string = ''
+param fwSubnetId string
+param fwManagementSubnetId string = ''
+param vnetAksSubnetAddressPrefix string
+param certManagerFW bool = false
+param acrPrivatePool bool = false
+param acrAgentPoolSubnetAddressPrefix string = ''
+param availabilityZones array = []
+param fwSku string
+
+var firewallPublicIpName = 'pip-afw-${resourceName}'
+var firewallManagementPublicIpName = 'pip-mgmt-afw-${resourceName}'
+
+var managementIpConfig = {
+  name: 'MgmtIpConf'
+  properties: {
+    publicIPAddress: {
+      id: !empty(fwManagementSubnetId) ? fwManagementIp_pip.id : null
+    }
+    subnet:{
+      id: !empty(fwManagementSubnetId) ? fwManagementSubnetId : null
+    }
+  }
+}
+
+resource fw_pip 'Microsoft.Network/publicIPAddresses@2021-03-01' = {
+  name: firewallPublicIpName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  zones: !empty(availabilityZones) ? availabilityZones : []
+  properties: {
+    publicIPAllocationMethod: 'Static'
+    publicIPAddressVersion: 'IPv4'
+  }
+}
+
+resource fwManagementIp_pip 'Microsoft.Network/publicIPAddresses@2021-03-01' = if(fwSku=='Basic') {
```

Comment:
last comment on case...just check all file names, var names, param names, etc ot make sure consistent w/ rest of core